allensdk/brain_observatory/behavior/running_processing.py

```diff
@@ -14,40 +16,378 @@ def calc_deriv(x, time):
+    ----------
+    vsig: Iterable (array-like)
+        The raw voltage data from the rotary encoder
+    vmax: Optional[float] (default=None)
+        The value at which, upon passing this threshold, the voltage
+         "wraps" back to 0V on the encoder.
+    max_threshold: float (default=5.1)
+        The maximum threshold for the `vmax` value. Used only if
+        `vmax` is `None`. To account for the possibility of outlier
+        data/encoder errors, the computed `vmax` should not exceed
+        this value.
+    max_diff: float (default=1.0)
+        The maximum voltage difference allowed between two adjacent
+        points, after accounting for the voltage "wrap". Values
+        exceeding this threshold will be set to np.nan.
+    Returns
+    -------
+    np.ndarray
+        1d np.ndarray of the "unwrapped" signal from `vsig`.
+    """
+    if not isinstance(vsig, np.ndarray):
+        vsig = np.array(vsig)
+    if vmax is None:
+        vmax = vsig[vsig < max_threshold].max()
+    unwrapped_diff = np.zeros(vsig.shape)
+    vsig_last = _shift(vsig)
+    if len(pos_wrap_ix):
+        # positive wraps: subtract from the previous value and add vmax
+        unwrapped_diff[pos_wrap_ix] = (
+            (vsig[pos_wrap_ix] + vmax) - vsig_last[pos_wrap_ix])
+    # negative: subtract vmax and the previous value
+    if len(neg_wrap_ix):
+        unwrapped_diff[neg_wrap_ix] = (
+            vsig[neg_wrap_ix] - (vsig_last[neg_wrap_ix] + vmax))
+    # Other indices, just compute straight diff from previous value
+    wrap_ix = np.concatenate((pos_wrap_ix, neg_wrap_ix))
+    other_ix = np.array(list(set(range(len(vsig_last))).difference(wrap_ix)))
+    unwrapped_diff[other_ix] = vsig[other_ix] - vsig_last[other_ix]
+    # Correct for wrap artifacts based on allowed `max_diff` value
+    # (fill with nan)
+    # Suppress warnings when comparing with nan values to reduce noise
+    with np.errstate(invalid='ignore'):
+        unwrapped_diff = np.where(
+            np.abs(unwrapped_diff) <= max_diff, unwrapped_diff, np.nan)
+    # Get nan indices to propogate to the cumulative sum (otherwise
+    # treated as 0)
+    unwrapped_nans = np.array(np.isnan(unwrapped_diff)).nonzero()
+    summed_diff = np.nancumsum(unwrapped_diff) + vsig[0]    # Add the baseline
+    summed_diff[unwrapped_nans] = np.nan
+    return summed_diff
+
+
+def _zscore_threshold_1d(data: np.ndarray,
+                         threshold: float = 5.0) -> np.ndarray:
+    """
+    Replace values in 1d array `data` that exceed `threshold` number
+    of SDs from the mean with NaN.
+    Parameters
+    ---------
+
+    Returns
+    -------
+    np.ndarray (1d)
+        A copy of `data` with values exceeding `threshold` SDs from
+        the mean replaced with NaN.
+    """
+    corrected_data = data.copy().astype("float")
+    scores = zscore(data, nan_policy="omit")
+    # Suppress warnings when comparing to nan values to reduce noise
+    with np.errstate(invalid='ignore'):
```

Comment:
I don't think this is necessary. I think this would suffice:
```
corrected_data[scores > threshold] = np.nan
```
```
In [15]: data = np.array([1.0, 1.2, 1.1, np.nan, 2.2, 1.05, 1.2, 1.01, 1.01])

In [16]: scores = zscore(data, nan_policy="omit")

In [17]: data[scores > 2.0] = np.nan

In [18]: data
Out[18]: array([1.  , 1.2 , 1.1 ,  nan,  nan, 1.05, 1.2 , 1.01, 1.01])
```

Reply:
I will have to check, but the issue is when you try to assess whether a `NaN` element is greater or less, etc. then numpy raises a warning. This warning isn't really useful for us and Doug said it was too noisy, so I'm suppressing them with this with context.

Reply:
I simplified the other line but the with context is still necessary to suppress the warning.

Reply:
ok